### ISSUE TITLE
can't validate sha256 from adfs

### ISSUE THREAD

**Issue by dcripplinger** [rohe/pysaml2#405]
Hi, this might be a long shot to figure out, but I'm desperate to find an answer somewhere. For the longest time, we've used pysaml2 and we've never been able to get it to work with idps using adfs unless we tell them to use sha1. Now we have someone that demands to use sha256 with their adfs. We have others that do sha256 with us successfully but not with adfs. We simply haven't been able to figure out why adfs sha256 fails signature validation. Does this sound at all similar to any known issues?

**Comment by jkakavas** [comment]
Hi,

It's not pretty clear from your description which software provides the signature, and which fails to validate it. Can you provide is with : 
 - Sample signed document
 - Error messages

**Comment by dcripplinger** [comment]
Good news. We figured it out.

First, to clarify your questions, the IdP (using ADFS) is supplying a signature to the SP (which is us) to use to verify saml responses. The problem was fixed when we inspected the embedded cert in saml responses after they switched to sha256. It wasn't the same as the cert they originally provided us. Since this has happened in every case we've ever tried to do this with an ADFS customer, our hypothesis is that ADFS updates with a new cert when someone changes their signing algorithm, and the users of ADFS are none the wiser. After we updated our records with the new cert, it worked fine.
